crates/pure-stage/src/stagegraph.rs

```diff
@@ -158,19 +194,24 @@ pub trait StageGraph {
     ///     }
     /// }
     /// ```
-    fn stage<Msg, St, F, Fut>(
+    ///
+    /// The returned [`StageRef`] can be used to send messages to this stage from other stages.
+    ///
+    fn register<Msg, St>(
         &mut self,
-        name: impl AsRef<str>,
-        f: F,
-    ) -> StageBuildRef<Msg, St, Self::RefAux<Msg, St>>
+        stage: &StageStateRef<Msg, St>,
+        stageable: impl Stage<Msg, St> + 'static + Send + Clone,
```

Comment:
What is gained by requiring an instance of a trait here? This means adding a `struct` that has no other use than to temporarily hold the initial state.